### PR TITLE
Documentation Clarity

### PR DESCRIPTION
**Pull request by skeet70**
I've had a few people at work get confused by the structure of the pureconfig documentation. Issues addressed here:

* Assumption that "Table of Contents" applied only to the README after clicking the first few links.
* Confused by "Whence the config files?"
* The `use` section showed a lot of magic without links to explanation. 
* The `example` section was an example of `parseString`, which is not the recommended way of using the library. The README usage/example should be the preferred way, so someone could could conceivably get to using it immediately.
* Fixed a broken link in the doc discussing how to use `loadConfigFromFiles`.

### CODE REVIEW

**Comment by ruippeixotog** [comment]
Hi @skeet70, thank you for the contribution! I like the changes you propose; I agree that our documentation needs some work and every contribution helps.

We use [tut](https://github.com/tpolecat/tut) for generating our README and the rest of the documentation and our build checks that the "source code" of the documentation is in sync with the generated docs. Because of that, changes to the README must be done in [core/src/main/tut](https://github.com/pureconfig/pureconfig/tree/master/core/src/main/tut) and changes to other docs should be done in [docs/src/main/tut](docs/src/main/tut). Afterwards, just run `sbt tut` and SBT will generate the "real" documentation. Be aware that code inside tut code blocks is actually compiled and run, so any config loading must point to real files.

The reorganization of the Table of Contents will most probably conflict with https://github.com/pureconfig/pureconfig/pull/319 which also simplifies that, so I guess it's better to leave it for later. Other than that, this looks good :)

**Comment by skeet70** [comment]
Ah, was wondering how the build had failed on doc changes. Thanks, I definitely didn't notice `tut` was involved, but it is a great tool. I'm pushing up a new commit generated using that.

**Review by melrief — APPROVED** [review]

**Comment by melrief** [comment]
Hi @skeet70, thank you for the PR! LGTM for me, I found the changes well done and the resulting README more readable. I'll wait for another review before merge.

File: core/src/main/tut/README.md
Comment (ruippeixotog) [inline core/src/main/tut/README.md:141]: My only concern with this particular change is that it's not being run by tut anymore. While I understand that loading a config from a file is much more useful than from a string for showcase purposes, this will get outdated if we ever change the `loadConfig` signature or the behavior of one of the readers.

I'm willing to let this without tut as it's a very short code section and we shouldn't change such an entrypoint to our library anyway. @melrief, do you agree?

Comment (melrief) [reply core/src/main/tut/README.md:141]: Most of the people use pureconfig to load from `application.conf` and I think @skeet70 changes are good. I'd like to have `tut` to check also this example but we can look into this later. Maybe there is way to still use `tut` somehow. For now, let's keep it unchecked @ruippeixotog.

**Review by ruippeixotog — APPROVED** [review]

**Pull request merged**
